src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,101 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elements.length) {
+            Object[] newObject = new Object[(int) (elements.length * 1.5)];
+            for (int i = 0; i < elements.length; i++) {
+                newObject[i] = elements[i];
+            }
+            elements = newObject;
+        }
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Can't write an element with index "
+                    + index + " to ArrayList");
+        }
+        if (size == elements.length) {
+            Object[] newObject = new Object[(int) (elements.length * 1.5)];
+            for (int i = 0; i < elements.length; i++) {
+                newObject[i] = elements[i];
+            }
+            elements = newObject;
+        }
```

Comment:
here you need to use `resize` method and replace all magic numbers with constants